api/src/main/java/io/opentelemetry/api/common/Attributes.java

```diff
@@ -20,16 +23,25 @@
  * <p>Null keys will be silently dropped.
  *
  * <p>Note: The behavior of null-valued attributes is undefined, and hence strongly discouraged.
+ *
+ * <p>Implementations of this interface *must* be immutable and have well-defined value-based
+ * equals/hashCode implementations. If an implementation does not strictly conform to these
+ * requirements, behavior of the OpenTelemetry APIs and default SDK cannot be guaranteed.
+ *
+ * <p>For this reason, it is strongly suggested that you use the implementation that is provided
+ * here via the factory methods and the {@link AttributesBuilder}.
  */
 @SuppressWarnings("rawtypes")
 @Immutable
-public abstract class Attributes extends ImmutableKeyValuePairs<AttributeKey, Object>
-    implements ReadableAttributes {
-  private static final Attributes EMPTY = Attributes.builder().build();
+public interface Attributes extends ReadableAttributes {
 
   @AutoValue
   @Immutable
-  abstract static class ArrayBackedAttributes extends Attributes {
+  abstract class ArrayBackedAttributes extends ImmutableKeyValuePairs<AttributeKey, Object>
```

Comment:
This class is public - can you move it to a top level package private class?

Reply:
ooh. good point. yes, will do so.

Reply:
done